daemon/response.go

```diff
@@ -270,6 +270,115 @@ func (f fileResponse) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	http.ServeFile(w, r, string(f))
 }
 
+type followChangeJSON struct {
+	Id      string `json:"id"`
+	Kind    string `json:"kind"`
+	Summary string `json:"summary"`
+	Ready   bool   `json:"ready"`
+	Status  string `json:"status"`
+	// XXX: do we want/need this?
+	OldStatus string `json:"old-status"`
+	// XXX: spawn/ready time?
+}
+
+type followChangesSeqResponse struct {
+	activeChanges map[*state.Change]state.Status
+
+	chgsCh chan *followChangeJSON
+}
+
+func newFollowChangesSeqResponse(st *state.State) *followChangesSeqResponse {
+	rsp := &followChangesSeqResponse{
+		activeChanges: make(map[*state.Change]state.Status),
+		// XXX: use a small buffer size
+		chgsCh: make(chan *followChangeJSON, 10),
+	}
+	st.AddTaskStatusChangedObserver(rsp.onTaskChanged)
+
+	return rsp
+}
+
+// XXX: better name?
+func (fc *followChangesSeqResponse) Stop() {
+	close(fc.chgsCh)
+}
+
+func (fc *followChangesSeqResponse) onTaskChanged(t *state.Task, oldTaskStatus, newTaskStatus state.Status) {
+	chg := t.Change()
+	if chg == nil {
+		return
+	}
+	new := chg.Status()
+	old, ok := fc.activeChanges[chg]
+	if !ok {
+		old = state.DefaultStatus
+	}
+	if new.Ready() {
+		delete(fc.activeChanges, chg)
+	} else {
+		fc.activeChanges[chg] = new
+	}
+
+	// XXX: do we need more filtering here? there is a lot of
+	// Doing->Done->Doing->Done right now
+	if new != old {
+		theChg := followChangeJSON{
+			Id:        chg.ID(),
+			Kind:      chg.Kind(),
+			Summary:   chg.Summary(),
+			Status:    new.String(),
+			OldStatus: old.String(),
+			Ready:     chg.IsReady(),
+		}
+
+		// ensure to not block, drop changes notifications if needed
+		select {
+		case fc.chgsCh <- &theChg:
+		default:
+			logger.Noticef("cannot write to full channel, dropping state change for id %v", chg.ID())
+		}
+	}
+}
+
+func (fc *followChangesSeqResponse) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json-seq")
+
+	flusher, hasFlusher := w.(http.Flusher)
+
+	var err error
+	writer := bufio.NewWriter(w)
+	enc := json.NewEncoder(writer)
+	for {
+		// XXX: provide a way to close this channel/exit this loop
+		chg := <-fc.chgsCh
```

Comment:
You can use the two value form to observe `chgsCh` being closed. Then you can just send actual change instances instead of just pointers.